### PR TITLE
use this.req insteadof this.ctx.req

### PR DESCRIPTION
**Pull request by hkc452**
use this.req directly

### CODE REVIEW

File: lib/response.js [hunk 1]
```diff
@@ -177,8 +177,10 @@ module.exports = {
     }
 
     // json
-    this.remove('Content-Length');
+    val = JSON.stringify(val);
```
Comment (dead-horse) [inline lib/response.js]: don't move these code to here. it will do stringify every time set `res.body`

File: lib/application.js [hunk 2]
```diff
@@ -229,10 +229,5 @@ function respond(ctx) {
   if ('string' == typeof body) return res.end(body);
   if (body instanceof Stream) return body.pipe(res);
 
-  // body: json
-  body = JSON.stringify(body);
-  if (!res.headersSent) {
-    ctx.length = Buffer.byteLength(body);
-  }
   res.end(body);
```
Comment (gabel0287) [inline lib/application.js]: What about this? 

```
if (body instance of Stream) return body.pipe(res);

return res.end(body);
```

**Comment by jonathanong** [comment]
what's the purpose of this?

**Comment by fl0w** [comment]
I'm going to close this PR due to inactivity.

**Pull request closed without merging**